### PR TITLE
feat: voice message support

### PR DESCRIPTION
**Pull request by Lulalaby**
Discord PR: https://github.com/discord/discord-api-docs/pull/6082

Implementation Reference: DisCatSharp

As sending is not supported, we won't implement it yet.

### CODE REVIEW

File: discord/permissions.py [hunk 1]
```diff
@@ -610,6 +610,14 @@ def moderate_members(self) -> int:
         """
         return 1 << 40
 
+    @flag_value
+    def send_voice_messages(self) -> int:
+        """:class:`bool`: Returns ``True`` if a umember can send voice messages.
```
Comment (JustaSqu1d) [inline discord/permissions.py:615]:
```suggestion
        """:class:`bool`: Returns ``True`` if a member can send voice messages.
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: discord/permissions.py (hunk 1 above)
Comment (Lulalaby) [reply discord/permissions.py:615]: Thanks!

**Review by Dorukyum — CHANGES_REQUESTED** [review]
The "versionadded"s should be 2.5

File: discord/permissions.py [hunk 2]
```diff
@@ -610,6 +610,14 @@ def moderate_members(self) -> int:
         """
         return 1 << 40
 
+    @flag_value
+    def send_voice_messages(self) -> int:
+        """:class:`bool`: Returns ``True`` if a member can send voice messages.
+
+        .. versionadded:: 2.4
```
Comment (Lulalaby) [inline discord/permissions.py:617]:
```suggestion
        .. versionadded:: 2.5
```

File: discord/message.py [hunk 3]
```diff
@@ -169,6 +169,16 @@ class Attachment(Hashable):
...
+
+    duration_secs: Optional[:class:`float`]
+        The duration of the audio file (currently for voice messages).
+
+        .. versionadded:: 2.4
+
+    waveform: Optional[:class:`str`]
+        The base64 encoded bytearray representing a sampled waveform (currently for voice messages).
+
+        .. versionadded:: 2.4
```
Comment (Lulalaby) [inline discord/message.py:181]:
```suggestion
        .. versionadded:: 2.5
```

File: discord/message.py [hunk 4]
```diff
@@ -169,6 +169,16 @@ class Attachment(Hashable):
         The attachment's description.
 
         .. versionadded:: 2.0
+
+    duration_secs: Optional[:class:`float`]
+        The duration of the audio file (currently for voice messages).
+
+        .. versionadded:: 2.4
```
Comment (Lulalaby) [inline discord/message.py:176]:
```suggestion
        .. versionadded:: 2.5
```

File: discord/flags.py [hunk 5]
```diff
@@ -404,6 +404,14 @@ def suppress_notifications(self):
 
         return 4096
 
+    @flag_value
+    def is_voice_message(self):
+        """:class:`bool`: Returns ``True`` if this message is a voice message.
+
+        .. versionadded:: 2.4
```
Comment (Lulalaby) [inline discord/flags.py:411]:
```suggestion
        .. versionadded:: 2.5
```

**Review by JustaSqu1d — APPROVED** [review]

**Review by BobDotCom — DISMISSED** [review]

**Review by BobDotCom — COMMENTED** [review]
test

**Review by NeloBlivion — APPROVED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Comment by Lulalaby** [comment]
oops

**Review by NeloBlivion — DISMISSED** [review]

File: discord/types/message.py [hunk 6]
```diff
@@ -66,6 +66,8 @@ class Attachment(TypedDict):
     size: int
     url: str
     proxy_url: str
+    duration_secs: NotRequired[float | None]
```
Comment (Dorukyum) [inline discord/types/message.py:69]: Why is this nullable while `waveform` isn't?

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Pull request merged**